derive_client/_bridge/client.py

```diff
@@ -324,173 +358,163 @@ def deposit_drv(self, amount: int) -> BridgeTxResult:
             message_id = source_event["args"]["msgId"]
         except Exception as e:
-            msg = f"Failed to retrieve `msgId` from the Socket MessageOutbound event log from source tx_receipt: {e}"
-            source_tx.exception = ValueError(msg)
-            return tx_result
+            raise BridgeEventParseError(f"Could not decode Socket MessageOutbound event: {e}") from e
 
         print(f"🔖 Source [{tx_result.source_chain.name}] MessageOutbound msgId: {message_id.hex()}")
-        target_event = target_socket.events.ExecutionSuccess()
-        filter_params = target_event._get_event_filter_params(
-            fromBlock=tx_result.target_from_block, abi=target_event.abi
+        filter_params = context.target_event._get_event_filter_params(
+            fromBlock=tx_result.target_from_block, abi=context.target_event.abi
         )
 
         def matching_message_id(log: AttributeDict) -> bool:
-            try:
-                decoded = target_event.process_log(log)
-                return decoded["args"].get("msgId") == message_id
-            except Exception:
-                return False
+            decoded = context.target_event.process_log(log)
+            return decoded.get("args", {}).get("msgId") == message_id
 
-        print(f"🔍 Listening for ExecutionSuccess on [{tx_result.target_chain.name}] at {target_socket.address}")
-        return wait_for_event(target_w3, filter_params, condition=matching_message_id)
+        print(f"🔍 Listening for ExecutionSuccess on [{tx_result.target_chain.name}] at {context.target_event.address}")
+        return wait_for_event(context.target_w3, filter_params, condition=matching_message_id)
 
     def poll_bridge_progress(self, tx_result: BridgeTxResult) -> BridgeTxResult:
-        # TODO: handle non-pending status
-
-        source_tx = tx_result.source_tx
-        target_tx = tx_result.target_tx
-
-        source_w3 = get_w3_connection(tx_result.source_chain)
-        target_w3 = get_w3_connection(tx_result.target_chain)
-
-        # 1. Timeout during source_tx.tx_receipt
-        if not source_tx.tx_receipt:
-            print(f"⏳ Checking source chain [{tx_result.source_chain.name}] tx receipt for {source_tx.tx_hash}")
-            source_tx.tx_receipt = wait_for_tx_receipt(w3=source_w3, tx_hash=source_tx.tx_hash)
-
-        # 2. Timeout waiting for event_log on target chain
-        if not target_tx.tx_hash:
-            match tx_result.bridge:
-                case BridgeType.SOCKET:
-                    event_log = self.fetch_socket_event_log(tx_result)
-                case BridgeType.LAYERZERO:
-                    event_log = self.fetch_lz_event_log(tx_result)
-                case _:
-                    raise ValueError()
-            target_tx.tx_hash = event_log["transactionHash"].to_0x_hex()
+        if tx_result.status is not TxStatus.PENDING:
+            raise AlreadyFinalizedError(f"Bridge already in final state: {tx_result.status.name}")
```

Comment:
if not pending, then it's final (see [comment here](https://github.com/8ball030/derive_client/pull/59/files#r2211674821)). We violently raise to let caller know (instead of returning their input directly).